docs/src/imagevolume_extensions.md

```diff
@@ -0,0 +1,332 @@
+# Image Volume Extensions
+<!-- SPDX-License-Identifier: CC-BY-4.0 -->
+
+CloudNativePG supports the **dynamic loading of PostgreSQL extensions** into a
+running `Cluster` using the [Kubernetes `ImageVolume` feature](https://kubernetes.io/docs/tasks/configure-pod-container/image-volumes/)
+and the `extension_control_path` GUC introduced in PostgreSQL 18, to which this
+project contributed.
+
+This feature allows you to mount a [PostgreSQL extension](https://www.postgresql.org/docs/current/extend-extensions.html),
+packaged as an OCI-compliant container image, as a read-only and immutable
+volume inside a running pod at a known filesystem path. You can then make the
+extension available to a PostgreSQL database for the `CREATE EXTENSION` command
+using the [`Database` resource’s declarative extension management](declarative_database_management.md/#managing-extensions-in-a-database).
+
+## Benefits
+
+Image volume extensions decouple the distribution of PostgreSQL operand
+container images from the distribution of extensions. This eliminates the
+need to define and embed extensions at build time within your PostgreSQL
+images—a major adoption blocker for PostgreSQL as a containerized workload,
+including from a security and supply chain perspective.
+
+As a result, you can:
+
+- Use the [official PostgreSQL `minimal` operand images](https://github.com/cloudnative-pg/postgres-containers?tab=readme-ov-file#minimal-images)
+  provided by CloudNativePG.
+- Dynamically add the extensions you need to your `Cluster` definitions,
+  without rebuilding or maintaining custom PostgreSQL images.
+- Reduce your operational surface by using immutable, minimal, and secure base
+  images while adding only the extensions required for each workload.
+
+Extension images must be built according to the
+[documented specifications](#image-specifications).
+
+## Requirements
+
+To use image volume extensions with CloudNativePG, you need:
+
+- **PostgreSQL 18 or later**, with support for `extension_control_path`.
+- **Kubernetes 1.33**, with the `ImageVolume` feature gate enabled.
+- **CloudNativePG-compatible extension container images**, ensuring:
+    - Matching PostgreSQL major version of the `Cluster` resource.
+    - Compatible operating system distribution of the `Cluster` resource.
+    - Matching CPU architecture of the `Cluster` resource.
+
+## How it works
+
+Extension images are defined in the `.spec.postgresql.extensions` stanza of a
+`Cluster` resource, which accepts an ordered list of extensions to be added to
+the PostgreSQL cluster.
+
+!!! Info
+    For field-level details, see the
+    [API reference for `ExtensionConfiguration`](cloudnative-pg.v1.md#postgresql-cnpg-io-v1-ExtensionConfiguration).
+
+Each image volume is mounted at `/extensions/<EXTENSION_NAME>`.
```

Comment:
PostgreSQL will not find extensions at that path, because it always wants some directory in the path to be named "postgresql" or "pgsql". We may want to consider, using `/extensions/postgresql/<EXTENSION_NAME>` or some such.